pipeline/extract/chat.py

```diff
@@ -0,0 +1,105 @@
+from typing import List, Set
+
+import pandas as pd
+import prefect
+from prefect import task
+
+from pipeline.types import (
+    ChatData,
+    ChatMatchId,
+    Community,
+    Match,
+    MatchMetadata,
+    Message,
+    User,
+    UserId,
+)
+
+# TODO:
+# fix naming conventions
+
+# add extract_chat_data in imports in flows/matching.py
+# call your funciton in the flow after the other extract tasks
+
+# Later: Write a function to join user objects and chatdata using user.uid
+# Function to join chatdata and all messages relating to that chat
+
+
+@task
+def extract_recent_chatdata(db, community: Community) -> List[ChatData]:
+    logger = prefect.context.get("logger")
+    chat_ref = db.reference(f"matches/{community}")
+    user_chat_records = chat_ref.get()
+    if not user_chat_records:
+        logger.info(f"No user-chat records for commmnity: {community}")
+        return []
+    logger.info(f"Extracted {len(user_chat_records)} user-chat records.")
+    ids: Set[ChatMatchId] = set()
+    chatList: List[ChatData] = []
+
+    for key in user_chat_records.keys():
+        if user_chat_records[key]["id"] not in ids:
+
+            timestamp = user_chat_records[key]["release_timestamp"]
+            id = user_chat_records[key]["id"]
+            participants = user_chat_records[key]["participants"]
+            logger.info(f"participants: {participants}")
+            rtag = user_chat_records[key]["release_tag"]
+            chatdata_title = user_chat_records[key]["title"]
+            raw_metadata = user_chat_records[key].get("metadata", {})
+            parsed_md = MatchMetadata(**raw_metadata)
+
+            # run extract_message
+            ext_message = extract_message_data(
+                db, community=community, chatid=id
+            )
+            if ext_message:
+                logger.info(f"Extracted message: {ext_message}")
+
+            cd1 = ChatData(
+                release_timestamp=timestamp,
+                chat_match_id=id,
+                community_id="sds",
+                participants=participants,
+                release_tag=rtag,
+                title=chatdata_title,
+                metadata=parsed_md,
+                messages=[],
```

Comment:
Similar issue here, we are expecting `ChatData` to contain a list of all the `Message` objects. You did a good job writing a separate method to extract messages. I like that approach because it's easier to test and understand.

But, we do need to populate the messages on the `ChatData` objects, so how will you do it? Check out my previous comment about the `User` objects needed in `participants` for ideas and ask @notedwin for advice.

Reply:
Used the extract message function to populate the messages in the `ChatData` objects.